BJS2-70591

SAM19991 opened the pull request.

SAM19991 committed:
BJS2-70591

MarselYsup commented on src/main/java/school/faang/bjs270591/Notification.java:
```diff
@@ -0,0 +1,16 @@
+    private final NotificationType type;
+    private final String message;
+
+    public Notification(NotificationType type, String message) {
```
можешь воспользоваться аннотацией Lombok ```@RequiredArgsConstructor```, такой способ будет более гибким, ведь если появятся новые поля, тебе не придется переписывать конструктор

MarselYsup commented on src/main/java/school/faang/bjs270591/Notification.java:
```diff
@@ -0,0 +1,16 @@
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
```
Здесь не нужен логгер, ты же ничего не логируешь в классе

MarselYsup commented on src/main/java/school/faang/bjs270591/NotificationManager.java:
```diff
@@ -0,0 +1,26 @@
+
+    public void registerHandler(NotificationType type, Consumer<Notification> handler) {
+        consumerMap.put(type, handler);
+        log.info("Registered handler for " + type);
```
воспользуйся форматированием 
```java
log.info("Registered handler for {}", type)
```

MarselYsup commented on src/main/java/school/faang/bjs270591/NotificationManager.java:
```diff
@@ -0,0 +1,26 @@
+
+    public void sendNotification(Notification notification) {
+        for (Map.Entry<NotificationType, Consumer<Notification>> entry : consumerMap.entrySet()) {
+            log.info("Sending notification " + entry.getKey());
```
воспользуйся форматированием, так не будут создаваться лишние строки в памяти StringPool

MarselYsup commented on src/main/java/school/faang/bjs270591/NotificationManager.java:
```diff
@@ -0,0 +1,26 @@
+    }
+
+    public void sendNotification(Notification notification) {
+        for (Map.Entry<NotificationType, Consumer<Notification>> entry : consumerMap.entrySet()) {
```
тебе не нужно проходиться по всем элементам маппы, нужно вытащить из нее по типу ```notification.getType()``` определенный ```Consumer``` и вызвать его

MarselYsup commented on src/main/java/school/faang/bjs270591/Main.java:
```diff
@@ -0,0 +1,28 @@
+        NotificationManager notificationManager = new NotificationManager();
+
+        notificationManager.registerHandler(NotificationType.EMAIL,
+                notification -> System.out.println("Email: " + notification.getMessage())
```
Используй тут  и ниже форматирование через printf, чтобы не создать большое кол-во объектов в StringPool(так как String неизменяемый и каждое складывание строк создает в памяти новую строку), а также такой подход более производительный, его проще поддерживать, особенно в случае сложных строк или когда выходные данные включают несколько типов данных

```java
System.out.printf("Email: %s", notification.getMessage())
```

MarselYsup commented on src/main/java/school/faang/bjs270591/Main.java:
```diff
@@ -0,0 +1,28 @@
+        notificationManager.sendNotification(emailNotification);
+        notificationManager.sendNotification(smsNotification);
+        notificationManager.sendNotification(pushNotification);
+
```
пустая строчка в конце метода не нужна

MarselYsup commented on src/main/java/school/faang/bjs270591/NotificationManager.java:
```diff
@@ -0,0 +1,26 @@
+@Slf4j
+public class NotificationManager {
+
+    private final Map<NotificationType, Consumer<Notification>> consumerMap = new HashMap<>();
```
Лучше назвать ```handlers``` или ```notificationHandlers``` - такое название лучше определяет эту мапу

SAM19991 committed:
Main.java:

Replaced string concatenation with System.out.printf for formatted output.

Notification.java:

Removed unnecessary logger.

Replaced the explicit constructor with Lombok’s @RequiredArgsConstructor.

NotificationManager.java:

Renamed the map to notificationHandlers.

Updated logger messages to use placeholder formatting.

Modified sendNotification to retrieve the notificationHandlers by notification type instead of iterating over the entire map.
